src/runtime/pay-flow-test.js

```diff
@@ -282,6 +287,49 @@ describes.realWin('PayCompleteFlow', {}, env => {
         expect(completeStub).to.be.calledOnce;
       });
     });
+
+    it('should start flow on correct payment response as encrypted obj', () => {
+      const encryptedResponse = {
+        redirectEncryptedCallbackData: INTEGR_DATA_STRING,
+        environment: 'PRODUCTION',
+      };
+      const result = new ActivityResult(ActivityResultCode.OK,
+          encryptedResponse,
+          'POPUP', location.origin, true, true);
+      const xhrFetchStub = sandbox.stub(Xhr.prototype, 'fetch',
+        () => Promise.resolve({json:() => Promise.resolve(INTEGR_DATA_OBJ_DECODED)}))
+      sandbox.stub(port, 'acceptResult', () => Promise.resolve(result));
+      const completeStub = sandbox.stub(PayCompleteFlow.prototype, 'complete');
+      PayCompleteFlow.configurePending(runtime);
+      return startCallback(port).then(() => {
+        expect(startStub).to.be.calledOnce;
+        expect(triggerPromise).to.exist;
+        return triggerPromise;
+      }).then(response => {
+        expect(response).to.be.instanceof(SubscribeResponse);
+        expect(response.purchaseData.raw).to.equal('{"orderId":"ORDER"}');
+        expect(response.purchaseData.signature).to.equal('PD_SIG');
+        expect(response.userData.idToken).to.equal(EMPTY_ID_TOK);
+        expect(JSON.parse(response.raw)).to.deep
+            .equal(JSON.parse(atob(INTEGR_DATA_STRING))['swgCallbackData']);
+        expect(completeStub).to.not.be.called;
+        response.complete();
+        expect(completeStub).to.be.calledOnce;
+
+        // Verify xhr call
+        expect(xhrFetchStub).to.be.calledOnce;
+        expect(xhrFetchStub).to.be.calledWith(
+          'https://pay.google.com/gp/p/apis/buyflow/process', ({
```

Comment:
Can you also add a test for a non-prod URL?

Reply:
Done